src/charm.py

```diff
@@ -327,12 +327,14 @@ def _prometheus_config(self) -> str:
             "scrape_timeout": "5s",
             "metrics_path": "/metrics",
             "honor_timestamps": True,
+            "honor_labels": True,
             "scheme": "http",
             "static_configs": [{"targets": [f"localhost:{self._port}"]}],
         }
         prometheus_config["scrape_configs"].append(default_config)  # type: ignore
         scrape_jobs = self.metrics_consumer.jobs()
         for job in scrape_jobs:
+            job["honor_labels"] = True
```

Comment:
Should the provider be allowed to set it to False?
Also I'd add a comment here that this is a deviation from prometheus defaults.

Reply:
No, see https://github.com/canonical/prometheus-k8s-operator/pull/219#discussion_r796530679 :-)